source/gameengine/Ketsji/BL_BlenderShader.cpp

```diff
@@ -153,8 +155,15 @@ void BL_BlenderShader::Update(RAS_MeshSlot *ms, RAS_Rasterizer *rasty)
 	RAS_MeshUser *meshUser = ms->m_meshUser;
 	const float (&obcol)[4] = meshUser->GetColor().Data();
 
+	float objectInfo[3];
+	if (GPU_get_material_builtins(m_gpuMat) & GPU_OBJECT_INFO) {
+		objectInfo[0] = meshUser->GetPassIndex();
+		objectInfo[1] = matPassIndex;
+		objectInfo[2] = BLI_hash_int_2d((uintptr_t)meshUser->GetClientObject(), 0) / ((float)0xFFFFFFFF);
+	}
```

Comment:
Store this value in RAS_MeshUser as the client will not change and this value too.